clusterloader/framework/rc_utils.go

```diff
@@ -0,0 +1,103 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	kutils "k8s.io/kubernetes/test/utils"
+)
+
+// CreateRC will create a new RC if it does not exist, or it will update an existing RC with a new replica count if it does exist
+func CreateRC(f *framework.Framework, namespace, name, imageName string, label map[string]string, replicas int) error {
+	rc, _ := f.ClientSet.Core().ReplicationControllers(namespace).Get(name, metav1.GetOptions{})
+	if rc.Name == "" {
+		newRC := newRC(name, int32(replicas), label, name, imageName)
+		_, err := createNewRCWithRetries(f, name, namespace, newRC)
+		if err != nil {
+			return err
+		}
+	} else {
+		_, err := framework.UpdateReplicationControllerWithRetries(f.ClientSet, namespace, name, func(update *v1.ReplicationController) {
+			x := int32(replicas)
+			update.Spec.Replicas = &x
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	podSelector := labels.SelectorFromSet(labels.Set(label))
+	// Wait for pods running matching label using podstore, does not take replica count into effect, nor owner reference
+	err := kutils.WaitForPodsWithLabelRunning(f.ClientSet, namespace, podSelector)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNewRCWithRetries uses polling to retry RC creation
+func createNewRCWithRetries(f *framework.Framework, name, namespace string, rcObj *v1.ReplicationController) (*v1.ReplicationController, error) {
+	var rc *v1.ReplicationController
+	var updateErr error
+	pollErr := wait.PollImmediate(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
+		var err error
+		if rc, err = f.ClientSet.Core().ReplicationControllers(namespace).Create(rcObj); err == nil {
+			framework.Logf("Created replication controller %q", name)
+			return true, nil
+		}
+		updateErr = err
+		return false, nil
+	})
+	if pollErr == wait.ErrWaitTimeout {
+		pollErr = fmt.Errorf("Unable to create new RC %q: %v", name, updateErr)
+	}
+	return rc, pollErr
+}
+
+// newRC creates a new ReplicationController config object
+func newRC(rsName string, replicas int32, rcPodLabels map[string]string, imageName string, image string) *v1.ReplicationController {
+	zero := int64(0)
+	return &v1.ReplicationController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: rsName,
+		},
+		Spec: v1.ReplicationControllerSpec{
+			Replicas: func(i int32) *int32 { return &i }(replicas),
+			Template: &v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: rcPodLabels,
+				},
+				Spec: v1.PodSpec{
```

Comment:
The downside is that then we have another external file dependency on the pod file. I will put a condition in both to check if the user has specified a podfile or just an imagename, and then create the appropriate struct.